pkgs/development/tools/vcpkg/default.nix

```
@@ -90,37 +35,44 @@ stdenv.mkDerivation rec {
       out = placeholder "out";
     in
     ''
-      #!${runtimeShell}
-
-      # make one root directory per vcpkg version
-      vcpkg_hash=$(echo -n "${out}" | sha256sum | cut -f1 -d ' ')
-      vcpkg_root_path="$HOME/.local/vcpkg/roots/$vcpkg_hash"
-
-      # create the root directory if it doesn't exist, and create symlinks to the vcpkg data
-      if [[ ! -d $vcpkg_root_path ]]; then
-        mkdir -p $vcpkg_root_path
-
-        ln -s ${out}/share/vcpkg/{docs,ports,scripts,triplets,versions,LICENSE.txt} $vcpkg_root_path/
-        ln -s ${out}/bin/vcpkg $vcpkg_root_path/
-
-        # this file is used as a lock by vcpkg, so it needs to be writable
-        touch $vcpkg_root_path/.vcpkg-root
-      fi
-
-      # add a special flag to tell the user where the root is. This can be used by nix-shell for
-      # finding the cmake toolchain file
-      if [[ "$1" == "--root-for-nix-usage" ]]; then
-        echo "$vcpkg_root_path"
-        exit 0
-      fi
+    #!${runtimeShell}
+    vcpkg_root_path="$HOME/.vcpkg/root/"
+
+    if [[ ! -d "$vcpkg_root_path" ]]; then
+      mkdir -p "$vcpkg_root_path"
+    fi
+    
+    if [[ ! -f "$vcpkg_root_path/.vcpkg-root" ]]; then
+      touch "$vcpkg_root_path/.vcpkg-root"
+    fi
+
+    if [[ ! -f "$vcpkg_root_path/vcpkg.disable-metrics" ]]; then
+      touch $out/share/vcpkg/vcpkg.disable-metrics
```

```suggestion
      touch $out/share/vcpkg/vcpkg.disable-metrics
      touch $vcpkg_root_path/vcpkg.disable-metrics
```

You can't write to out in shell-script on runtime

---

Hey thanks for those suggestions. This is really helpful!